### ISSUE TITLE
Which Iteration to use?

### ISSUE THREAD

**Issue by dk683** [alexgkendall/SegNet-Tutorial#24]
Hello, @alexgkendall.
Thanks to your guidance, I successfully ran SegNet with out custom dataset.
Looking into outputs, our network seemed to converge after 110 iterations.

If I understood correctly, I understood that your SegNet also converged around 100 iterations.
However, the tutorial uses iteration 10000 for testing SegNet.

I would like to ask your advice about which iterations I should be using for testing performance.
Thank you!

**Comment by alexgkendall** [comment]
Fantastic. Do you mean 110 epochs? You should periodically test your network on a validation set, and chose the best performing model. Somewhere around 100 epochs sounds right.

**Comment by dk683** [comment]
Yes, I meant epochs :-) Thank you for advice!

**Comment by isn4** [comment]
I'm sorry I'm not too sure I'm understanding.
One epoch is [#training images]/[batch size] iterations correct?
As in for the CamVid segnet tutorial one epoch completed after (367/1) iterations, yes?
And thus segnet would only converge at 100 epochs or 36,700 iterations.
So why use iteration 10000 when it seems like the net has not yet converged?
Perhaps my understanding of epochs and iterations is incorrect: http://stackoverflow.com/questions/4752626/epoch-vs-iteration-when-training-neural-networks
